File: forge/src/main/java/dev/franckyi/mcfx/forge/MCFXModForge.java
package dev.franckyi.mcfx.forge;

import dev.architectury.platform.forge.EventBuses;
import dev.franckyi.mcfx.MCFXMod;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;

@Mod(MCFXMod.MOD_ID)
public class MCFXModForge {
    public MCFXModForge() {
        // Submit our event bus to let architectury register our content on the right time
        EventBuses.registerModEventBus(MCFXMod.MOD_ID, FMLJavaModLoadingContext.get().getModEventBus());
        MCFXMod.init();
    }
}

File: forge/src/main/kotlin/forge/MCFXModForge.kt
package forge

import dev.architectury.platform.forge.EventBuses
import dev.franckyi.mcfx.MCFXMod
import net.minecraftforge.api.distmarker.Dist
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.loading.FMLLoader
import thedarkcolour.kotlinforforge.forge.MOD_BUS

@Mod(MCFXMod.MOD_ID)
object MCFXModForge {
    init {
        EventBuses.registerModEventBus(MCFXMod.MOD_ID, MOD_BUS)
        if (FMLLoader.getDist() == Dist.CLIENT) {
            MCFXMod.init()
        }
    }
}